### ISSUE TITLE
FileRotationLogger crashes when archiving full log files

### ISSUE THREAD

**Issue by tucknology** [sushichop/Puppy#10]
Hi!

In `FileRotationLogger.swift` the function `rotateFiles()` fails at line 109 during the following instruction:

```swift
try FileManager.default.moveItem(at: fileURL, to: archivedFileURL)
```

This happens because in `Formatter.swift` the function ` dateFormatter(_:, locale:, dateFormat:, timeZone:)` creates a timestamp that includes colons (`:`) which the file system does not like. 

One potential solution is changing the `dateFormat` in line 13 of `Formatter.swift` from `"yyyy-MM-dd'T'HH:mm:ss.SSSZZZZZ"` to `"yyyy-MM-dd'T'HH-mm-ss.SSS"`. This removes the explicit colons in the original date format string, as well as an implicit colon which is added by `ZZZZZ` which expands to an `HH:MM` style presentation of the timezone. 

You might also consider reducing the complexity of this string further, since I'd argue if a detailed timestamp is necessary then it can be placed **inside** the log file, rather than in the file name.

One alternative could be to structure like this:

```
myLog.log
myLog.log.1
myLog.log.2
```

This proposal would simplify the implementation of the rest of the `rotateFiles` method that is concerned with removing old archived files.

**Comment by sushichop** [comment]
@tucknology 
Thanks for reporting this. I decided not to use colon. 
And for now I still want to handle the case someone wants to check the date without opening the file. 
So I'll use ISO8601 basic format (no-hyphen and GMT) instead of extended format and this fixes the issue with small chaneg.
See #11 .

> One alternative could be to structure like this:
> 
> ```
> myLog.log
> myLog.log.1
> myLog.log.2
> ```

This is another good idea!
I'll also consider this solution in the future. It needs more big change in the code.
And I have created this issue in order to keep in mind (#12).

**Comment by sushichop** [comment]
Since this issue was fixed by #11, I'll close it.
Feel free to reopen it or create a new issue if you will have any questions.

Thanks.

**Comment by sushichop** [comment]
> One alternative could be to structure like this:
> 
> ```
> myLog.log
> myLog.log.1
> myLog.log.2
> ```

I have adde this feature. See #32.

Thanks.

**Comment by sushichop** [comment]
I have released Puppy@0.4.0 that fixes this issue.
Check [release note](https://github.com/sushichop/Puppy/releases/tag/0.4.0) and [README](https://github.com/sushichop/Puppy#readme) because some specifications of `FileLogger` and `FileRotationLogger` have been changed.

Thanks.
